client/service/src/lib.rs

```diff
@@ -322,17 +324,7 @@ impl<TBl: Unpin, TCl, TSc: Unpin, TNetStatus, TNet, TTxPool, TOc> Future for
 		}
 
 		while let Poll::Ready(Some(task_to_spawn)) = Pin::new(&mut this.to_spawn_rx).poll_next(cx) {
-			// TODO: Update to tokio 0.2 when libp2p get switched to std futures (#4383)
-			let executor = tokio_executor::DefaultExecutor::current();
-			use futures01::future::Executor;
-			if let Err(err) = executor.execute(task_to_spawn.unit_error().compat()) {
-				debug!(
-					target: "service",
-					"Failed to spawn background task: {:?}; falling back to manual polling",
-					err
-				);
-				this.to_poll.push(Box::pin(err.into_future().compat().map(drop)));
-			}
+			tokio::spawn(task_to_spawn);
```

Comment:
This will unfortunately crash on wasm-browser.
Browser support is the reason why we check for errors and push on `this.to_poll`.

Reply:
Do I push the `JoinHandle` to `to_poll`?

Reply:
No. `to_poll` is about running futures locally. The problem is that `tokio` does not work at all from within the browser.

Reply:
I'd prefer to keep it the same way as it is currently is, except that we need to wait for https://github.com/tokio-rs/tokio/pull/2118

Reply:
I'm happy to wait.